### ISSUE TITLE
Weird behavior with bun and next build

### ISSUE THREAD

**Issue by gfffrtt** [prisma/prisma#22444]
### Bug description

Hello everyone, I'm currently build a NextJs application with bun and turbopack, and I'm deploying it with docker, I have 2 dockerfiles: Local.Dockerfile and Production.Dockerfile a .dockerignore and a package.json (obviously, lol) I'll paste them below so you can replicate the error on your machine.

**What the error actually is**: When i run the `Local.Dockerfile` I get not problems it generates the prisma client, and it's all good. The problem arrives when I try to build the `Production.Dockerfile` which after running the same command as I run the `Local.Dockerfile`, and trying to build, the compiler says that the prisma client doesn't exist, and I should consider running prisma generate (lol). I don't know for sure if this is a problem with prisma so if nothing pans out here I might as well report this problem to NextJs then Bun. Or maybe I've just made a sily mistake, anyways thank you, if you need more info let me know in the comments. 

`Local.Dockerfile`
```
FROM oven/bun:1 as base
COPY --from=node:18 /usr/local/bin/node /usr/local/bin/node

WORKDIR /app

COPY package*.json ./
COPY bun.lockb ./

RUN bun install

COPY . .

ENV DATABASE_URL postgresql://root:root@app-mentallize-db-1:5432/app-mentallize?schema=public
ENV NODE_ENV development

RUN bun db:generate

EXPOSE 3000

CMD ["bun", "dev"]
```

`Production.Dockerfile`
```
FROM oven/bun:1 as base
FROM base AS deps

WORKDIR /app

COPY package.json bun.lockb ./

RUN bun install --frozen-lockfile

FROM base AS builder

WORKDIR /app

COPY --from=deps /app/node_modules ./node_modules
COPY . .

RUN bun run db:generate
RUN bun run build

FROM base AS runner

WORKDIR /app

ENV NODE_ENV production
ENV DATABASE_URL [ommitted]

RUN addgroup --system --gid 1001 nodejs
RUN adduser --system --uid 1001 nextjs

COPY --from=builder /app/public ./public

RUN mkdir .next
RUN chown nextjs:nodejs .next

COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./
COPY --from=builder --chown=nextjs:nodejs /app/.next/static ./.next/static

USER nextjs

EXPOSE 3000/tcp

ENV PORT 3000
ENV HOSTNAME "0.0.0.0"

CMD ["bun", "run", "server.js"]
```

`.dockerignore`
```
node_modules
Dockerfile*
docker-compose*
.dockerignore
.gitignore
Makefile
```

`package.json`
```
{
  "private": true,
  "trustedDependencies": [
    "@prisma/client"
  ],
  "scripts": {
    "dev": "next dev --turbo",
    "build": "next build",
    "start": "next start",
    "db:generate": "bunx prisma generate",
    "db:push": "bunx prisma db push"
  },
  "dependencies": {
    "@prisma/client": "5.7.0",
    "next": "latest",
    "prisma": "^5.7.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@types/node": "20.8.10",
    "@types/react": "18.2.33",
    "@types/react-dom": "18.2.14",
    "typescript": "^5.2.2"
  }
}
```

### How to reproduce

1. Initialize a NextJs project with turbopack using bun (bun create next-app --example with-turbopack)
2. Copy my Production.Dockerfile and package.json, so you have the same enviroment as me
3. Run docker

### Expected behavior

As said above, the expected behavior should be a flawless build of the Production.Dockerfile, because in my Local.Dockerfile I do essentially the same thing but with a different result, it actually works.

### Prisma information

```prisma
datasource db {
    provider = "postgresql"
    url      = env("DATABASE_URL")
}

generator client {
    provider = "prisma-client-js"
}

model User {
    id       String  @id @default(cuid())
    email    String  @unique @db.VarChar(150)
    password String  @db.VarChar(150)
    staff    Staff?
    client   Client?

    @@map("user")
}

enum ROLE {
    OWNER
    EMPLOYEE
}

enum GROUP {
    SCOPED
    GLOBAL
}

model Staff {
    id     String @id @default(cuid())
    role   ROLE
    group  GROUP
    userId String @unique
    user   User   @relation(fields: [userId], references: [id])

    @@map("staff")
}

enum GENDER {
    MALE
    FEMALE
}

model Client {
    id       String @id @default(cuid())
    fullName String @db.VarChar(200)
    cpf      String @db.VarChar(11)
    age      Int    @db.SmallInt
    gender   GENDER
    userId   String @unique
    user     User   @relation(fields: [userId], references: [id])

    @@map("client")
}

```

Nothing really important my problem isn't with the code, as I've said above, this error occurs when building
```ts
import { PrismaClient } from "@prisma/client";
import { randomUUID } from "crypto";

export default async function Page() {
  const db = new PrismaClient();

  await db.user.create({
    data: { email: `${randomUUID()}@gmail.com`, password: randomUUID() },
  });
  const users = await db.user.findMany();

  return (
    users && (
      <div>
        {users.map((user) => (
          <p>
            #{user.id}: {user.email}
          </p>
        ))}
      </div>
    )
  );
}
```

### Environment & setup

- OS: Unbuntu 22.04
- Database: PostgreSQL
- Node.js version: 18 (but my bun version for the runtime is 1.0)

### Prisma Version

```
prisma                  : 5.7.0
@prisma/client          : 5.7.0
Computed binaryTarget   : debian-openssl-3.0.x
Operating System        : linux
Architecture            : x64
Node.js                 : v20.10.0
Query Engine (Node-API) : libquery-engine 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.7.0-41.79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Default Engines Hash    : 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Studio                  : 0.495.0
```

**Comment by cholasimmons** [comment]
Prisma sort of depends on npm under the hood I think, it's most likely working locally because you have that installed there.
It took me 2 frustrating weeks to figure that out.
Also Prisma@5.7.0 seems to have a new issue for me (I'm using it on ElysiaJS [with Bun JS]) so I had to step back to 5.6.0

**Comment by gfffrtt** [comment]
@cholasimmons by locally I mean, development, because it's a dockerfile, I only called it Local.Dockerfile because it seemed prettier than Development.Dockefile (I know seems stupid but).

But I'll definitly try out changing from 5.7.0 to 5.6.0

**Comment by gfffrtt** [comment]
Hello guys, I'm closing this issue I've found the error, after looking over my Dockerfiles, I've noticed that in `Local.Dockerfile` I have the command `COPY --from=node:18 /usr/local/bin/node /usr/local/bin/node` which essentially install node on that building stage so prisma can use it, whereas in my `Production.Dockerfile` I don't have it in my builder stage. I guess it was a dumb little mistake like I said. Thank you @cholasimmons for contributing!
